doc/TWAMP/SAI-Proposal-TWAMP-LIGHT.md

```diff
@@ -0,0 +1,1085 @@
+
+Unlike the TWAMP, TWAMP Light is a simplified version of TWAMP where the full TWAMP protocol is not necessary. The main differences between TWAMP and TWAMP Light are that TWAMP Light does not include complex authentication mechanisms, and does not require a control connection to be established before test packets can be sent.  So TWAMP Light is a lightweight protocol which is easy to deploy and use.
+
+## TWAMP Light Solution ##
+TWAMP Light does not require a control connection to set up and manage the measurement sessions. Instead, it uses a simple message exchange process between two endpoints to initiate and terminate the measurements. In hence, the roles of Control-Client, Server and Session-Sender are implemented in one host referred to as the controller, and the role of Session-Reflector is implemented in another host referred to as the responder as described in RFC-5357.  TWAMP Light is a Two-way measurement witch is common in IP networks, primarily because synchronization between local and remote clocks is unnecessary for round-trip delay, and measurement support at the remote end may be limited to a simple echo function.
+![Workflow](./figures/TWAMP_Light_role.png "Figure 1: TWAMP Workflow")
+To calculate the latency, jitter, and packet loss rate of network, the following steps are typically followed:
+
+- Session-Sender as the active endpoint of the TWAMP Light sends a Test-request packet with a tx timestamp as t0.
+- Session-Reflector receives the Test-request packet and captures a rx timestamp as t1.
+- Prior to the transmission of the Test-response packet, Session-Reflector captures a tx timestamp as t2.
+- Session-Reflector encodes t1 and t2 into the Test-response packet, then sends it out.
+- Session-Sender receives the Test-response packet and captures a rx timestamp as t3.
+
+The algorithm is:
+
+- Latency = (t3-t0) – (t2-t1)
+- Jitter = | Latency1 – Latency0 |
+- Packet loss rate = (txPkt – rxPkt) / tx
+
+## TWAMP Light offloading in ASIC ##
+Since TWAMP Light is a simple and lightweight protocol, it is easy for vendors to implement TWAMP Light in their ASIC, which can improve performance and reduce CPU overload on network devices. TWAMP Light offloading in ASIC should work for sending and receiving TWAMP Light packets, collecting measurement data such as latency, jitter and packet loss. And also, RFC-5357 TWAMP Light should be followed. For TWAMP Light hardware solution, the main requirements are as follows:
+* Can be deployed as one sender
+  * Construct Test-request packets based on configuration including IP header, UDP port, etc.
+  * Transmit Test-request packets to a reflector within tx timestamps which are captured in ASIC.
```

Comment:
"within" or "with" ?

Reply:
I think it is "within" because it means that the timestamp should be encoded in packets.